sdk/src/main/java/io/horizen/account/utils/EthereumTransactionEncoder.java

```diff
@@ -64,9 +61,9 @@ private static RlpList LegacyTransaction2RlpList(EthereumTransaction tx, boolean
             result.add(RlpString.create(s));
         } else {
             if (tx.isEIP155()) {
-                result.add(RlpString.create(EthereumTransactionUtils.convertToBytes(tx.getChainId())));
-                result.add(RlpString.create(new byte[] {}));
-                result.add(RlpString.create(new byte[] {}));
+                result.add(RlpString.create(BigInteger.valueOf(tx.getChainId())));
+                result.add(RlpString.create(BigInteger.ZERO));
```

Comment:
just a double check: are we sure that ZERO byte representation is empty array?

Reply:
Yes, in that case the `create() `method returns `new RlpString(EMPTY)`